cicecore/cicedyn/infrastructure/ice_domain.F90

```diff
@@ -216,21 +216,36 @@ subroutine init_domain_blocks
    call broadcast_scalar(maskhalo_bound,    master_task)
    call broadcast_scalar(add_mpi_barriers,  master_task)
    call broadcast_scalar(debug_blocks,      master_task)
-   if (my_task == master_task) then
-     if (max_blocks < 1) then
-       max_blocks=( ((nx_global-1)/block_size_x + 1) *         &
-                    ((ny_global-1)/block_size_y + 1) - 1) / nprocs + 1
-       max_blocks=max(1,max_blocks)
-       write(nu_diag,'(/,a52,i6,/)') &
-         '(ice_domain): max_block < 1: max_block estimated to ',max_blocks
-     endif
-   endif
    call broadcast_scalar(max_blocks,        master_task)
    call broadcast_scalar(block_size_x,      master_task)
    call broadcast_scalar(block_size_y,      master_task)
    call broadcast_scalar(nx_global,         master_task)
    call broadcast_scalar(ny_global,         master_task)
 
+   ! Set nprocs if not set in namelist
```

Comment:
Is it better to compute this on all procs or to compute it on master proc then broadcast?  I'm not sure one is better than the other, but we certainly want to make sure each proc has the same value, so maybe we should do it before the broadcast?

Also, why did you revert the improved max_blocks computation back to the original version?  If your implementation is more robust, let upgrade, even if it might change again.

Reply:
Part A: This was just a stylistic decision. It seems easier to read / follow / modify if all the broadcast scalar calls together. The result is the same.

Part B: Oh I misread / I misunderstood. Ok the max_blocks using proc_decomposition is back in :)